tools/CheckH5.py

```diff
@@ -0,0 +1,207 @@
+                keys_difference = test_keys ^ expected_keys
+                self.unit_test.assertEqual(
+                    keys_difference, {},
+                    "test and expected h5 groups must have identical" +
+                    "subgroups and data members. Found differences: " +
+                    str(keys_difference) + "\nin comparing group " +
+                    test_entity + " to group " + expected_entity)
+                for key in test_keys:
+                    checks_passed = checks_passed and self.check_h5_file(
+                        h5_file, test_entity + "/" + key,
+                        expected_entity + "/" + key)
+            else:
+                for key in test_keys:
+                    checks_passed = checks_passed and self.check_h5_file(
+                        h5_file, test_entity + "/" + key, None)
+            return checks_passed
+
+    # Apply the h5 check to every h5 file within the `run_directory`.
+    # If neither the expected nor test objects are present in an h5 file, that
+    # file is skipped. However, if none of the h5 files have the requested
+    # datasets, the test fails.
+    # It is not supported for the `run_directory` to have h5 files from
+    # multiple runs.
+    def perform_check(self, run_directory):
+        checks_performed = False
+        for filename in os.listdir(run_directory):
+            if filename.endswith(".h5"):
+                # It is recommended to include subTest commands once
+                # python3 can be required:
+                # with self.unit_test.subTest(filename=filename):
+                with h5py.File(filename, 'r') as check_h5:
+                    if self.test_h5_entity in check_h5 or (
+                            self.expected_h5_entity is not None
+                            and self.expected_h5_entity in check_h5):
+                        checks_performed = True
+                        self.unit_test.assertTrue(
+                            self.test_h5_entity in check_h5)
+                        self.unit_test.assertTrue(
+                            self.expected_h5_entity is None
+                            or self.expected_h5_entity in check_h5)
+                        self.check_h5_file(check_h5, self.test_h5_entity,
+                                           self.expected_h5_entity)
+        self.unit_test.assertTrue(checks_performed)
+
+
+# The unit test object for performing all H5 checks for a given input file.
+class H5CheckTestCase(unittest.TestCase):
+
+    # The parameters for the H5 output checks are determined by the
+    # command-line arguments. The first argument (sys.argv[1]) is the yaml
+    # input file. The second argument is the directory in which to find the
+    # run's.h5 files.
+    #
+    # The H5 checks and arguments are parsed from the yaml file comments
+    # we require a format like:
+    # ```
+    # # H5Checks:
+    # #   - CheckLabel: "label"
+    # #     TestH5: "/h5_name.dat"
```

Comment:
We should be forced to specify a glob to match the file name. That would fix the current bug when running in parallel.